gson/src/main/java/com/google/gson/JsonParser.java

```diff
@@ -26,7 +26,47 @@
 
 /**
  * A parser to parse JSON into a parse tree of {@link JsonElement}s.
+ * 
+ * <p>Here's an example of parsing from a string:
+ * 
+ * <pre>
+ * String json = "{\"key\": \"value\"}";
+ * JsonElement jsonElement = JsonParser.parseString(json);
+ * JsonObject jsonObject = jsonElement.getAsJsonObject();
+ * </pre>
  *
+ * <p>It can also parse from a reader:
+ * 
+ * <pre>
+ * try (Reader reader = new FileReader("C:\\my_json.json")) {
+ *     JsonElement jsonElement = JsonParser.parseReader(reader);
+ *     JsonObject jsonObject = jsonElement.getAsJsonObject();
```

Comment:
Maybe to match the code style of this project and of the other code examples in the Javadoc, might be good to use only 2 spaces instead of 4 spaces for indentation.

(applies to the code snippets below as well)